### PR TITLE
Fix max iter

### PR DESCRIPTION
**Pull request by hndgzkn**
Fixes number of local segments and  local max_iter in compliance with alphacsc implementation

### CODE REVIEW

**Review by tomMoral — APPROVED** [review]
LGTM!

File: dicodile/utils/segmentation.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def compute_n_seg(self):
         self.n_seg_per_axis = []
         for size_ax, size_seg_ax in zip(self.signal_support, self.seg_support):
             # Make sure that n_seg_ax is of type int (and not np.int*)
-            n_seg_ax = max(1, int(size_ax // size_seg_ax))
+            n_seg_ax = max(1, int(size_ax // size_seg_ax - 1))
```
Comment (tomMoral) [inline dicodile/utils/segmentation.py:88]: Actually, are you sure this covers for all the coordinates?

Comment (hndgzkn) [reply dicodile/utils/segmentation.py]: I've updated segmentation in accordance with alphacsc. Could you review the recent changes?

Comment (hndgzkn) [reply dicodile/utils/segmentation.py]: This does not fix the issue, I need to work further on that. Please ignore my previous comment.

File: dicodile/workers/dicod_worker.py [hunk 2]
```diff
@@ -112,7 +113,7 @@ def compute_z_hat(self):
                 k0, pt0, dz = None, None, 0
 
             # update the accumulator for 'random' strategy
-            accumulator = max(abs(dz), accumulator)
+            accumulator += abs(dz)
```
Comment (tomMoral) [inline dicodile/workers/dicod_worker.py:116]: Why do you change this? It should not be a sum here but the max over all segments

File: dicodile/workers/dicod_worker.py [hunk 3]
```diff
@@ -747,7 +748,7 @@ def recv_signal(self):
         local_seg_support = None
         if self.n_seg == 'auto':
             n_seg = None
-            local_seg_support = 2 * np.array(atom_support) - 1
+            local_seg_support = 2 * np.array(atom_support) + 1
```
Comment (tomMoral) [inline dicodile/workers/dicod_worker.py:751]: why `+1`? I though it was `2*atoms_support` in `alphacsc`

Comment (hndgzkn) [reply dicodile/workers/dicod_worker.py:751]: But there are still other problems

File: dicodile/utils/segmentation.py [hunk 4]
```diff
@@ -85,7 +85,9 @@ def compute_n_seg(self):
         self.n_seg_per_axis = []
         for size_ax, size_seg_ax in zip(self.signal_support, self.seg_support):
             # Make sure that n_seg_ax is of type int (and not np.int*)
-            n_seg_ax = max(1, int(size_ax // size_seg_ax))
+            rem = size_ax % size_seg_ax
+            n_seg_ax = max(1, int(size_ax // size_seg_ax) +
+                           (1 if rem != 0 else 0))
```
Comment (tomMoral) [inline dicodile/utils/segmentation.py:90]:
```suggestion
                           ((size_ax % size_seg_ax) != 0)
```

File: dicodile/utils/segmentation.py [hunk 5]
```diff
@@ -85,7 +85,9 @@ def compute_n_seg(self):
         self.n_seg_per_axis = []
         for size_ax, size_seg_ax in zip(self.signal_support, self.seg_support):
             # Make sure that n_seg_ax is of type int (and not np.int*)
-            n_seg_ax = max(1, int(size_ax // size_seg_ax))
+            rem = size_ax % size_seg_ax
```
Comment (tomMoral) [inline dicodile/utils/segmentation.py:88]:
```suggestion
```

File: dicodile/utils/segmentation.py [hunk 6]
```diff
@@ -85,7 +85,8 @@ def compute_n_seg(self):
         self.n_seg_per_axis = []
         for size_ax, size_seg_ax in zip(self.signal_support, self.seg_support):
             # Make sure that n_seg_ax is of type int (and not np.int*)
-            n_seg_ax = max(1, int(size_ax // size_seg_ax))
+            n_seg_ax = max(1, int(size_ax // size_seg_ax) +
+                           ((size_ax % size_seg_ax) != 0)
```
Comment (tomMoral) [inline dicodile/utils/segmentation.py:89]:
```suggestion
                           (size_ax % size_seg_ax) != 0)
```

File: dicodile/utils/segmentation.py [hunk 7]
```diff
@@ -85,7 +85,8 @@ def compute_n_seg(self):
         self.n_seg_per_axis = []
         for size_ax, size_seg_ax in zip(self.signal_support, self.seg_support):
             # Make sure that n_seg_ax is of type int (and not np.int*)
-            n_seg_ax = max(1, int(size_ax // size_seg_ax))
+            n_seg_ax = max(1, int(size_ax // size_seg_ax) +
+                           (size_ax % size_seg_ax) != 0)
```
Comment (hndgzkn) [inline dicodile/utils/segmentation.py:89]:
```suggestion
                           ((size_ax % size_seg_ax) != 0))
```

**Review by hndgzkn — COMMENTED** [review]
I think we need to put paranthesis around. If not, it evaluates to 0 even if there is remainder.

**Comment by tomMoral** [comment]
Oups sorry, I went too fast ^^
This is why it is nice to have some tests :)

**Comment by hndgzkn** [comment]
> Oups sorry, I went too fast ^^ This is why it is nice to have some tests :)

Tests added and pass. Ok to merge!

**Review by tomMoral — APPROVED** [review]
LGTM! thx

**Review by agramfort — APPROVED** [review]
@hndgzkn 🙏

**Pull request merged**